delphi/umap_narrative/reset_conversation.py

```diff
@@ -38,6 +39,177 @@ def get_boto_resource(service_name: str):
     return boto3.resource(service_name, **resource_args)
 
 
+def delete_single_item(dynamodb, table_name, key_config):
+    """
+    Delete a single item from a DynamoDB table.
+
+    Args:
+        dynamodb: DynamoDB resource object
+        table_name: Name of the table
+        key_config: Dict with 'key_name' and 'key_value' for the item to delete
+
+    Returns:
+        Number of items deleted (0 or 1)
+    """
+    try:
+        table = dynamodb.Table(table_name)
+        table.delete_item(Key={key_config['key_name']: key_config['key_value']})
+        logger.info(f"  ✓ {table_name}: 1 item deleted.")
+        return 1
+    except Exception as e:
+        if 'ResourceNotFoundException' in str(e):
+            logger.debug(f"  - {table_name}: Table does not exist")
+            return 0
+        elif 'ConditionalCheckFailedException' in str(e):
+            logger.debug(f"  - {table_name}: Item did not exist")
+            return 0
+        else:
+            logger.error(f"  ✗ {table_name}: Error - {e}")
+            return 0
+
+def batch_delete_items(table, items, primary_keys):
+    """Helper to perform batch deletion and handle errors."""
+    if not items:
+        return 0
+    try:
+        with table.batch_writer() as batch:
+            for item in items:
+                key_to_delete = {pk: item[pk] for pk in primary_keys}
+                batch.delete_item(Key=key_to_delete)
+        logger.info(f"  ✓ {table.name}: {len(items)} items deleted.")
+        return len(items)
+    except Exception as e:
+        logger.error(f"  ✗ {table.name}: Batch delete failed - {e}")
+        return 0
+
+
+def _fetch_and_delete_items(dynamodb, table_name, key_config, operation_type, operation_kwargs):
+    """
+    Generic helper to fetch items from DynamoDB and delete them.
+
+    Args:
+        dynamodb: DynamoDB resource object
+        table_name: Name of the table
+        key_config: Dict with 'keys' (list of key names)
+        operation_type: 'query' or 'scan'
+        operation_kwargs: Kwargs for the query/scan operation
+
+    Returns:
+        Number of items deleted
+    """
+    try:
+        # Track timing for the operation
+        start_time = time.time()
+
+        operation_name = 'Query' if operation_type == 'query' else 'Scan'
```

Comment:
We cannot accept scans. Scan operations are the equivalent of `select *` on an entire table. All of these items have Global Secondary Indexes, meaning we can query them without having to scan every item in a table